packtools/sps/validation/errata.py

```diff
@@ -0,0 +1,150 @@
+from packtools.sps.validation.utils import format_response
+from packtools.sps.models.related_articles import RelatedItems
+from packtools.sps.models.article_dates import HistoryDates
+
+
+# Classe base para validação de artigos, implementando a lógica comum de validação de artigos relacionados
+class ValidationBase:
+    def __init__(self, xml_tree, related_article_type):
+        # Inicializa a árvore XML e configura o tipo de artigo relacionado a ser validado
+        self.xml_tree = xml_tree
+        self.article_lang = xml_tree.get(
+            "{http://www.w3.org/XML/1998/namespace}lang"
+        )  # Obtém o idioma do artigo
+        self.article_type = xml_tree.xpath(".")[0].get(
+            "article-type"
+        )  # Obtém o tipo de artigo
+        # Filtra os artigos relacionados com base no tipo especificado
+        self.related_articles = [
+            article
+            for article in RelatedItems(xml_tree).related_articles
+            if article.get("related-article-type") == related_article_type
+        ]
+
+    # Método principal de validação de artigos relacionados, que pode ser sobrescrito por subclasses
+    def validate_related_article(
+        self,
+        expected_related_article_type,
+        expected_message,
+        advice,
+        error_level="ERROR",
+    ):
+        if (
+            self.article_type == "correction"
+        ):  # Verifica se o tipo de artigo é uma correção
+            if (
+                not self.related_articles
+            ):  # Se não houver artigos relacionados, retorna um erro
+                yield self._format_error_response(
+                    expected=f'at least one <related-article related-article-type="{expected_related_article_type}">',
+                    advice=f'provide <related-article related-article-type="{expected_related_article_type}">',
+                    error_level=error_level,
+                )
+            else:
+                # Se houver artigos relacionados, valida cada um e retorna um sucesso
+                for article in self.related_articles:
+                    yield self._format_success_response(
+                        article, expected_message, error_level
+                    )
```

Comment:
@Rossi-Luciano tente usar a estrutura `for`+ `else`

```python
for n in range(2, 10):
    for x in range(2, n):
        if n % x == 0:
            print(n, 'equals', x, '*', n//x)
            break
    else:
        # loop fell through without finding a factor
        print(n, 'is a prime number')
```

Reply:
OK, @robertatakenaka